src/ast/nodes/MemberExpression.ts

```diff
@@ -2,53 +2,55 @@ import relativeId from '../../utils/relativeId';
 import { ExpressionNode, Node, NodeBase } from './shared/Node';
 import Variable from '../variables/Variable';
 import ExecutionPathOptions from '../ExecutionPathOptions';
-import { isLiteral } from './Literal';
 import CallOptions from '../CallOptions';
 import MagicString from 'magic-string';
-import Identifier, { isIdentifier } from './Identifier';
+import Identifier from './Identifier';
 import NamespaceVariable from '../variables/NamespaceVariable';
 import ExternalVariable from '../variables/ExternalVariable';
 import { ForEachReturnExpressionCallback, SomeReturnExpressionCallback } from './shared/Expression';
 import { NodeType } from './NodeType';
 import { NodeRenderOptions, RenderOptions } from '../../utils/renderHelpers';
-import { isUnknownKey, ObjectPath, ObjectPathKey, UNKNOWN_KEY } from '../values';
+import {
+	LiteralValueOrUnknown,
+	ObjectPath,
+	ObjectPathKey,
+	UNKNOWN_KEY,
+	UNKNOWN_VALUE
+} from '../values';
 import { BLANK } from '../../utils/blank';
+import Literal from './Literal';
 
-const validProp = /^[a-zA-Z_$][a-zA-Z_$0-9]*$/;
-
-function getPropertyKey(memberExpression: MemberExpression): ObjectPathKey {
+function getPropertyKey(memberExpression: MemberExpression): string | null {
 	return memberExpression.computed
 		? getComputedPropertyKey(memberExpression.property)
 		: (<Identifier>memberExpression.property).name;
 }
 
-function getComputedPropertyKey(propertyKey: ExpressionNode): ObjectPathKey {
-	if (isLiteral(propertyKey)) {
-		const key = String(propertyKey.value);
-		return validProp.test(key) ? key : UNKNOWN_KEY;
+function getComputedPropertyKey(propertyKey: ExpressionNode): string | null {
+	if (propertyKey instanceof Literal) {
+		return String(propertyKey.value);
 	}
-	return UNKNOWN_KEY;
+	return null;
```

Comment:
It seems like there is some duplication here between this and `getComputedKey`. Are the separate methods needed because the variable bindings aren't all available at the initialization time? If so perhaps we should defer the call to `getComputedPropertyKey` to when needed rather than init?

Reply:
The reason this function still exists (which is only used initially!) is that it ties into the namespace replacement logic and I did not want to dirty my hands on this just now. The same applies to `getPathIfNotComputed`.

On a somewhat related note: I was wondering if we should inline side-effect-free resolvable computed property keys. The upside is of course slightly better performance, the downside is that if the key is a very long string, the key might be duplicated many times.

Reply:
Interesting, which code exactly are you referring to here?

Inlining would be great - but I'd strongly suggest avoiding inlining anything whose contributing literals are also used elsewhere.